### DISCUSSION TITLE
How to implement drop-down list with search function?

### DISCUSSION THREAD

**Discussion by Amtech281** [firecmsco/firecms#392] [Q&A] [answered]
hi
We are referencing in one document other documents in another collection. this collection has several thousands entries. Therefore a dropdown is not convenient, is it possible to impelment in the create/edit form a searchable dropdown list, where I e..g enter "madri" and then all documents containing "madri" are show in the dropdown list?
thanks for any help

**Comment by fgatti675** [comment, marked as answer]
Hi @Andre3424 
You can integrate text search in the collection you are referencing.
Unfortunately this requires some backend adaptations on your side (since Firestore does not provide text search) besides the changes in the CMS:
https://firecms.co/docs/collections/text_search
We can support you with the implementation if you need :)

**Reply by Amtech281** [reply]
@fgatti675 thanks! We check it out and will be glad to contact you in case we need help
